File: app/src/main/java/com/example/fondodepantalla/FragmentosAdministrador/LocationHelper.java
package com.example.fondodepantalla.FragmentosAdministrador;

import android.annotation.SuppressLint;
import android.content.Context;
import android.location.Location;
import androidx.annotation.Nullable;
import com.google.android.gms.location.FusedLocationProviderClient;
import com.google.android.gms.location.LocationServices;

public class LocationHelper {

    private final FusedLocationProviderClient fusedLocationClient;

    public LocationHelper(Context context) {
        fusedLocationClient = LocationServices.getFusedLocationProviderClient(context);
    }

    @SuppressLint("MissingPermission")
    public void getCurrentLocation(OnLocationResultListener listener) {
        fusedLocationClient.getLastLocation()
                // 'location' aquí PUEDE ser nulo si no hay última ubicación conocida
                .addOnSuccessListener(location -> listener.onLocationResult(location))
                // Aquí envías 'null' explícitamente en caso de fallo
                .addOnFailureListener(e -> listener.onLocationResult(null));
    }

    public boolean isInsideCompanyArea(double currentLat, double currentLon) {
        float[] result = new float[1];
        Location.distanceBetween(
                currentLat, currentLon,
                Constants.COMPANY_LATITUDE, Constants.COMPANY_LONGITUDE,
                result
        );
        return result[0] <= Constants.ALLOWED_RADIUS_METERS;
    }

    public interface OnLocationResultListener {
        // 🌟 CORRECCIÓN: Añadir '@Nullable' para permitir valores nulos
        // Esto soluciona la incompatibilidad entre el listener y la implementación
        void onLocationResult(@Nullable Location location);
    }
}

File: app/src/main/java/com/example/fondodepantalla/Utils/LocationHelper.kt
package com.example.fondodepantalla.Utils // 1. Paquete 'Utils' (con mayúscula)

import android.annotation.SuppressLint
import android.content.Context
import android.location.Location
import com.google.android.gms.location.LocationServices

// 2. El 'context' se pasa al constructor primario
class LocationHelper(context: Context) {

    // 3. La propiedad se inicializa directamente
    private val fusedLocationClient = LocationServices.getFusedLocationProviderClient(context)

    /**
     * 4. La interfaz 'OnLocationResultListener' se reemplaza por un tipo de función: (Location?) -> Unit
     * Esto te permite llamar a la función usando un bloque lambda, que es mucho más limpio.
     */
    @SuppressLint("MissingPermission")
    fun getCurrentLocation(listener: (location: Location?) -> Unit) {
        fusedLocationClient.lastLocation
            .addOnSuccessListener { location -> listener(location) } // 'it' también funcionaría, pero 'location' es más explícito
            .addOnFailureListener { listener(null) }
    }

    /**
     * 5. Asegúrate de que tu archivo 'Constants' (o donde sea que
     * 'COMPANY_LATITUDE', 'COMPANY_LONGITUDE' y 'ALLOWED_RADIUS_METERS' vivan)
     * sea accesible desde este paquete (por ejemplo, importándolo).
     */
    fun isInsideCompanyArea(currentLat: Double, currentLon: Double): Boolean {
        // La creación de 'FloatArray' es un poco diferente en Kotlin
        val result = FloatArray(1)
        Location.distanceBetween(
            currentLat, currentLon,
            Constants.COMPANY_LATITUDE, // Asumo que 'Constants' está importado
            Constants.COMPANY_LONGITUDE,
            result
        )
        return result[0] <= Constants.ALLOWED_RADIUS_METERS
    }
}